### PR TITLE
Update tests and add manual tests for the enhanced CPU reservation feature

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#2207

### CODE REVIEW

**Comment by shuo-wu** [comment]
@khushboo-rancher @meldafrawi @joshimoo Can you take a look?

File: docs/content/manual/release-specific/v1.1.1/system-upgrade-with-deprecated-cpu-setting.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+## Related issue
+https://github.com/longhorn/longhorn/issues/2207
+
+## Test step
+1. Deploy a cluster that each node has different CPUs.
+2. Launch Longhorn v1.1.0.
+3. Deploy some workloads using Longhorn volumes.
+4. Upgrade to the latest Longhorn version. Validate the workloads work fine and no instance manager pod crash during the upgrade.
```
Comment (khushboo-rancher) [inline docs/content/manual/release-specific/v1.1.1/system-upgrade-with-deprecated-cpu-setting.md:12]: Will the existing value of `Guaranteed Engine Manager CPU` become the new value of `node.Spec.EngineManagerCPURequest` for all the nodes and `Guaranteed Replica Manager CPU` set to a default value? What would be the default value?

**Review by khushboo-rancher — DISMISSED** [review]
LGTM

File: docs/content/manual/release-specific/v1.1.1/system-upgrade-with-deprecated-cpu-setting.md (hunk 1 above)
Comment (shuo-wu) [reply docs/content/manual/release-specific/v1.1.1/system-upgrade-with-deprecated-cpu-setting.md:12]: `node.Spec.EngineManagerCPURequest` is a field in node spec, which is different from the setting `Guaranteed Engine Manager CPU`:
1. The setting will be ignored by the specific node only when its spec field `EngineManagerCPURequest` is not 0.
2. The setting is a percentage value. While the node field holds a milli value.

Comment (shuo-wu) [reply docs/content/manual/release-specific/v1.1.1/system-upgrade-with-deprecated-cpu-setting.md:12]: For more details, you can check the doc PR or the LEP PR.

**Comment by shuo-wu** [comment]
@joshimoo @meldafrawi Do you need to take a look? Since the related manager PR is already merged and there is an old setting deprecated, we need to checkout this PR ASAP in case of the test failures.

**Review by meldafrawi — APPROVED** [review]
LGTM

**Pull request merged**